lib/logging.cpp

```diff
@@ -100,13 +107,39 @@ std::istream& operator>>(std::istream& strm, severity_level& level) {
     return strm;
 }
 
+/// Custom formatter for escaped messages.
+///
+/// Not really clear but just a wrapper around the c_decor formatter.
+struct escaped_message_formatter {
+    explicit escaped_message_formatter(logging::attribute_name const& name) :
+        f_{expr::stream << expr::c_decor[expr::stream << expr::smessage]} {
+    }
+    void operator()(logging::record_view const& rec, logging::formatting_ostream& strm) const {
+        f_(rec, strm);
+    }
+
+private:
+    /// @brief The formatter itself.
+    boost::log::formatter f_;
+};
+
+/// The factory for the EscMessage formatter.
+struct escaped_message_formatter_factory : public logging::formatter_factory<char> {
+    formatter_type create_formatter(logging::attribute_name const& attr_name, args_map const& args) {
+        return formatter_type(escaped_message_formatter(attr_name));
+    }
+};
+
 void init(const std::string& logconf) {
     init(logconf, "");
 }
 
 void init(const std::string& logconf, std::string process_name) {
     BOOST_LOG_FUNCTION();
 
+    // First thing - register the custom formatter for EscMessage
+    logging::register_formatter_factory("EscMessage", boost::make_shared<escaped_message_formatter_factory>());
```

Comment:
I would call this "EscapedMessage", that's more in-line with the naming of Boost.Log and more readable anyway.